docs/content/templating_best_practices.md

```diff
@@ -0,0 +1,197 @@
+  },
+  "db_users": {
+    "dev1": "cloudprovisioner_iam",
+    "int2": "cloudprovisioner_iam",
+    "gov": "cloudprovisioner_iam",
+    "prod": "{{ index .TF_OUTPUTS_ALL_SERVICES \"cloud_provisioner\" \"db-user-name\" \"value\" }}"
+
+}
+```
+
+**values.gtmpl:**
+
+```gotemplate
+{{- defineDatasource "config" "/path/to/config/file" -}}
+{{- $config := ds "config" }}
+db:
+  auth: {{ index $config.db_auth .CLUSTER.name }}
+  user: {{ tpl (index $config.db_users .CLUSTER.name) . }}
+  name: {{ index .TF_OUTPUTS_ALL_SERVICES "cloud_provisioner" "db-name" "value" }}
+  region: {{ .CLUSTER.region }}
+```
+
+### Working with Conditionals
+
+Inline logic works well for simple booleans, but quickly becomes difficult to read and maintain.
+Instead, use variables:
+
+```gotemplate
+replicas: {{- if eq .ENVIRONMENT.name "prod" }} 3 {{ else }} 1 {{ end }}
+```
+
+becomes:
+
+```gotemplate
+{{- $replicas := 1 -}}
+{{- if eq .ENVIRONMENT.name "prod" }}{{- $replicas = 3 -}}{{- end }}
+---
+replicas: {{ $replicas }}
+```
+
+For more complex conditionals that you want to reuse across templates, consider using a datasource with the logic pre-computed:
+
+**config.yaml:**
+
+```yaml
+replicas:
+  dev: 1
+  staging: 2
+  prod: 3
+```
+
+**values.gtmpl:**
+
+```gotemplate
+{{- defineDatasource "config" "config.yaml" -}}
+{{- $config := ds "config" -}}
+replicas: {{ index $config.replicas .ENVIRONMENT.name }}
+```
+
+### Reducing Path Repetition
+
+When working with nested configurations, you might find yourself repeating the same path multiple times:
+
+```gotemplate
+
+envVars:
+  - name: REGION
+    value: {{ .CLUSTER.type.region }}
+  - name: PARTITION
+    value: {{ .CLUSTER.type.partition }}
+  - name: SERVICEDEPENDENCY_HOST
+    value: {{ .CLUSTER.type.dep.host }}
+  - name: SERVICEDEPENDENCY_PORT
+    value: {{ .CLUSTER.type.dep.port }}
+
+```
+
+Using `with` blocks cuts repetition, makes structure easier to read, and reduces chances of errors:
+
+```gotemplate
+
+{{- with .CLUSTER.type -}}
+envVars:
+  - name: REGION
+    value: {{ .region }}
+  - name: PARTITION
+    value: {{ .partition }}
+  - name: SERVICEDEPENDENCY_HOST
+    value: {{ .dep.host }}
+  - name: SERVICEDEPENDENCY_PORT
+    value: {{ .dep.port }}
+{{- end -}}
+
+```
+
+### Closing Thoughts
+
+This guide covered several key practices for writing maintainable Gomplate templates:
+
+- Use environment variables for high-level, shared configuration
+- Extract complex configuration into datasources instead of embedding conditionals
+- Keep conditionals clean by using variables or moving logic to datasources
+- Use `with` blocks to reduce repetitive path references
+
+Remember that templates should be:
+
+- Easy to read and maintain
+- Modular and reusable
+- Well-structured with configuration separate from logic
+- Consistent in their approach to similar problems
+
+While these patterns emerged from a Helm values use case, they apply to any complex templating needs with Gomplate.
```

Comment:
I don't think it's necessary to reference provenance here - this is good advice on its own! 

```suggestion
```